portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -31,12 +30,32 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  ArrayList<String> msglist = new ArrayList<String> (Arrays.asList("Hi!","Ni Hao!","Bonjour!","Hola!"));
-  
+  ArrayList<String> msglist = new ArrayList<String>();
+  String jsonMsg;
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("application/json;");
-    response.getWriter().println(new Gson().toJson(msglist));
+    response.getWriter().println(new Gson().toJson(msglist)) ;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String inputMsg = getParameter(request, "comment-input", "");
+    msglist.add(inputMsg);
+    response.sendRedirect("/index.html");
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```

Comment:
you can replace this by using the ternary operator. 
return request.parameter(name) == null ? defaultValue : value;

Reply:
Agreed with using ternary operator, but can probably keep:
String value = request.getParameter(name); and use value for the return statement.

Reply:
But what could be better is to move the logic of checking null above, as Warfa mentioned, if null, we may not want to add it to the list.

Reply:
To make better logic, I deleted the getParameter() method and directly check whether the string is empty to put it inside the arraylist